src/data/weather.ts

```diff
@@ -538,3 +544,49 @@ export const getWeatherConvertibleUnits = (
   hass.callWS({
     type: "weather/convertible_units",
   });
+
+export const getForecast = (
+  forecast?: ForecastAttribute[],
+  forecastDaily?: ForecastAttribute[],
+  forecastHourly?: ForecastAttribute[],
+  forecastTwiceDaily?: ForecastAttribute[],
+  forecastType?: string | undefined
+): ForecastAttribute[] | undefined => {
+  if (
+    forecastType === "daily" &&
+    forecastDaily?.length &&
+    forecastDaily?.length > 2
+  ) {
+    return forecastDaily;
+  }
+  if (
+    forecastType === "hourly" &&
+    forecastHourly?.length &&
+    forecastHourly?.length > 2
+  ) {
+    return forecastHourly;
+  }
+  if (
+    forecastType === "twice_daily" &&
+    forecastTwiceDaily?.length &&
+    forecastTwiceDaily?.length > 2
+  ) {
+    return forecastTwiceDaily;
+  }
+  if (forecastType === "legacy" && forecast?.length && forecast?.length > 2) {
+    return forecast;
+  }
+  if (forecastType === undefined) {
```

Comment:
Why isn't the `stateObj.attributes.forecast` part of the default? Legacy entities will not get a forecast now?